app/assets/javascripts/controllers/schedule/schedule_form_controller.js

```diff
@@ -381,15 +398,30 @@ ManageIQ.angular.app.controller('scheduleFormController', ['$http', '$scope', 's
     return $scope.scheduleModel.log_protocol === 'AWS S3';
   };
 
+  $scope.undefinedScheduleField = function(value) {
+    return (value === '' || typeof value === 'undefined');
+  };
+
   $scope.regionRequired = function() {
-    return ($scope.s3Backup() &&
-      ($scope.scheduleModel.log_aws_region === '' || typeof $scope.scheduleModel.log_aws_region === 'undefined'));
+    return ($scope.s3Backup() && $scope.undefinedScheduleField($scope.scheduleModel.log_aws_region));
+  };
+
+  $scope.swiftSecurityProtocolSelect = function() {
+    return $scope.scheduleModel.log_protocol === 'OpenStack Swift';
```

Comment:
This should be `return $scope.scheduleModel.action_typ == "db_backup" &&  $scope.scheduleModel.log_protocol === 'OpenStack Swift';` to prevent following error:
- Configuration -> Settings -> Schedules -> add/edit one
- Select `Action` to be `Database backup` and `Type` to be `OpenStack Swift`
- Select  `Action` to be `VM Analysis`

Form contains some options from previous step depending if they were filled or not.
<img width="1032" alt="screen shot 2018-10-04 at 1 45 54 pm" src="https://user-images.githubusercontent.com/9210860/46472558-e183a100-c7dd-11e8-8afe-883284b545a9.png">

Reply:
Done.